### DISCUSSION TITLE
ESP32-s3 no storage available

### DISCUSSION THREAD

**Discussion by steeley** [micropython/micropython#13577] [ESP32]
Hi, I have  an esp32-s3 4Meg module:
[(https://www.digikey.co.uk/en/products/detail/espressif-systems/ESP32-S3-WROOM-1U-N4R8/16162634](url)
and I can flash firmware to it which works, but there s no storage for my .py files.
I read the docs and went to compile a new firmware, which seems to build ok. I found a discussion here:
[https://github.com/orgs/micropython/discussions/10156](url)
 and I thinkI have edited the files correctly, but my module still has no storage.

I/m using Micopython 1.22.0
ESP IDF 5.1.2

I then tried using this module with Arduino and it works fine, so thats something. here is the output when an Arduino sketch boots on the module:
`ESP-ROM:esp32s3-20210327
Build:Mar 27 2021
rst:0x1 (POWERON),boot:0x8 (SPI_FAST_FLASH_BOOT)
SPIWP:0xee
mode:DIO, clock div:1
load:0x3fce3808,len:0x44c
load:0x403c9700,len:0xbe4
load:0x403cc700,len:0x2a38
entry 0x403c98d4

`

Any advice appreciated.

TIA

**Comment by peterhinch** [comment]
Have you tried the pre-built [generic S3 firmware](https://micropython.org/download/ESP32_GENERIC_S3/)?

**Reply by steeley** [reply]
thanks Peter - yes I have. it seems only for 8MB flash  modules, mine is 4MB flash 8MB SRAM. Unless I'm missing something?
